src/main/java/spoon/support/reflect/code/CtBlockImpl.java

```diff
@@ -163,79 +158,35 @@ private boolean shouldInsertAfterSuper() {
 	@Override
 	public <T extends CtStatementList> T setStatements(List<CtStatement> statements) {
 		if (statements == null || statements.isEmpty()) {
-			this.statements = CtElementImpl.emptyList();
+			this.statements.clear();
```

Comment:
We already call `clear` below in the method: this could now be simlified by just adding a if not null around the final `addAll`

Reply:
you are right